eng/common/templates/steps/run-pwsh-with-auth.yml

```
@@ -0,0 +1,38 @@
+parameters:
+- name: name
+  type: string
+  default: "RunPwshWithAuth"
+- name: displayName
+  type: string
+  default: "Run PowerShell with Auth"
+- name: serviceConnection
+  type: string
+  default: ""
+- name: command
+  type: string
+  default: null
+- name: continueOnError
+  type: boolean
+  default: false
+- name: dockerClientOS
+  type: string
+  default: "linux"
+- name: condition
+  type: string
+  default: true
+
+steps:
+- task: AzureCLI@2
+  name: ${{ parameters.name }}
+  displayName: ${{ parameters.displayName }}
```

Just an idea - maybe we should append something like "(Authenticated)" to the displayName here to make it easier to tell which pipeline task was chosen. Would require change to default displayName above.

```suggestion
  displayName: ${{ parameters.displayName }} (Authenticated)
```

---

Wouldn't internal runs always be authenticated and public runs would not?

---

> Wouldn't internal runs always be authenticated and public runs would not?

I guess that's not true. There would be scenarios for internal runs where no service connection is in scope.

---

I was just thinking about from a debugging point of view, it'd require less knowledge of how the tasks are structured to see if it's authenticating when you want it to.